Sources/SwiftExtensions/AsyncUtils.swift

```diff
@@ -165,6 +165,17 @@ extension Collection where Element: Sendable {
       count = indexedResults.count
     }
   }
+
+  /// Invoke `body` for every element in the collection and wait for all calls of `body` to finish
+  package func concurrentForEach(_ body: @escaping @Sendable (Element) async -> Void) async {
+    await withTaskGroup(of: Void.self) { taskGroup in
+      for element in self {
+        taskGroup.addTask {
+          await body(element)
+        }
+      }
+    }
```

Comment:
Doesn't this need either a wait on each of the tasks or a taskGroup.waitForAll()?

Reply:
No, from the documentation of `withTaskGroup`. I only learned this now, which is why I had `waitForAll` in older implementations.

> A group waits for all of its child tasks to complete or be canceled before it returns.